### PR TITLE
[WIP] AudioDSP core fixes

### PR DESCRIPTION
**Pull request by AchimTuran**
@AlwinEsch @FernetMenta 
These are my core fixes for AudioDSP and libKODI_audioengine.

Furthermore I added some small usability fixes for AudioDSP.

### CODE REVIEW

**Comment by AchimTuran** [comment]
jenkins build this please

**Comment by MartijnKaijser** [comment]
please use doxygen style for note and todo

**Comment by AchimTuran** [comment]
Ok will do that.

File: xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEBuffer.cpp [hunk 1]
```diff
@@ -123,6 +123,19 @@ bool CActiveAEBufferPool::Create(unsigned int totaltime)
   config.sample_rate = m_format.m_sampleRate;
   config.channel_layout = CAEUtil::GetAVChannelLayout(m_format.m_channelLayout);
 
+  // discard already created buffers
+  if (m_allSamples.size() > 0 || m_freeSamples.size() > 0)
```
Comment (FernetMenta) [inline xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEBuffer.cpp]: this may work but the method this is done here is not correct. buffers are only deleted by pushing them to the discarded queue. why is this needed?

File: xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEBuffer.cpp [hunk 2]
```diff
@@ -261,19 +284,31 @@ bool CActiveAEBufferPoolResample::Create(unsigned int totaltime, bool remap, boo
 
   if (m_useResampler || m_changeResampler)
   {
-    m_resampler = CAEResampleFactory::Create();
+    if(!m_resampler)
+      m_resampler = CAEResampleFactory::Create();
+
+    AEAudioFormat m_resamplerInFormat;
```
Comment (FernetMenta) [inline xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEBuffer.cpp]: variables on the stack never start with m_

File: xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEBuffer.cpp [hunk 3]
```diff
@@ -261,19 +284,31 @@ bool CActiveAEBufferPoolResample::Create(unsigned int totaltime, bool remap, boo
 
   if (m_useResampler || m_changeResampler)
   {
-    m_resampler = CAEResampleFactory::Create();
+    if(!m_resampler)
```
Comment (FernetMenta) [inline xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEBuffer.cpp]: apart from the missing space this does not look right. why is there already a resampler on create?

File: xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEBuffer.cpp (hunk 1 above)
Comment (AchimTuran) [inline xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEBuffer.cpp]: I kniw that it is not right, but is needed because adsp can change his format according to active or non avtive add-ons. The main problem is that AE doesn't get any response about changes from adsp. Consequently I jave to recreate the buffers internally.

File: xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEBuffer.cpp (hunk 2 above)
Comment (AchimTuran) [inline xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEBuffer.cpp]: Thx, right. It is a copy paste issue.

File: xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEBuffer.cpp (hunk 3 above)
Comment (AchimTuran) [inline xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEBuffer.cpp]: I mentioned this during a debugging session from adsp.
It's the same as above. AE has no feedback from adsp.

File: xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEBuffer.cpp (hunk 1 above)
Comment (FernetMenta) [inline xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEBuffer.cpp]: > The main problem is that AE doesn't get any response about changes from adsp

please fix this. re-configuration has to be done by Configure()

File: xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEBuffer.cpp (hunk 3 above)
Comment (FernetMenta) [inline xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEBuffer.cpp]: root cause needs to be fixed.

Comment (AchimTuran) [inline xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEBuffer.cpp]: Right, I will try to implement it in the next weeks.

**Comment by AchimTuran** [comment]
This PR gets closed. I created https://github.com/xbmc/xbmc/pull/10756 for further discussions.
